src/jestTestAdapters/JestTestAdapterFactory.ts

```diff
@@ -0,0 +1,37 @@
+import JestTestAdapter from './JestTestAdapter';
+import JestPromiseAdapter from './JestPromiseTestAdapter';
+import JestCallbackAdapter from './JestCallbackTestAdapter';
+import * as semver from 'semver';
+
+export default class JestTestAdapterFactory {
+  public static getJestTestAdapter(loader?: NodeRequire): JestTestAdapter {
+    let jestTestAdapter: JestTestAdapter;
+
+    switch (this.getJestType(loader)) {
```

Comment:
Do you really need to call that method? Now you're coupling two methods together with magic strings.

Reply:
No I don't need to call the method, should refactor that.